Neptune/Services/IService.cs

```diff
@@ -1,11 +1,31 @@
-﻿namespace Neptune
+﻿using System;
+using System.Net.Http;
+
+namespace Neptune
 {
     public interface IService { }
-    public interface IMessageService<IS>
+    public class MessageService<IS>
         where IS:IService
     {
-        R SendMessage<M, R>(M msg)
-            where M:IMessage<IS>
-            where R:IResponse<M>;
+        private HttpClient client;
+
+        public MessageService(HttpClient client)
+        {
+            this.client = client;
+        }
+
+        public R GetRequest<M, R>(M msg)
+            where M : IGetRequest<IS>
+            where R : IResponse<IS, M>
+        {
+            throw new NotImplementedException();
+        }
+
+        public R PostRequest<M, R>(M msg)
```

Comment:
Let's support PUT and DELETE

Reply:
I agree, so long as we can all agree on the general pattern of creating a message type, a message response type, and a service type all of which are bound together by the type checker.